pkg/dogstatsd/server.go

```diff
@@ -171,6 +172,9 @@ type Server struct {
 	originTelemetry bool
 
 	enrichConfig enrichConfig
+
+	// logger is an instant of the logger config that can be used to create new logger for dogstatsd-stats metrics
+	Logger slog.LoggerInterface
```

Comment:
```suggestion
	StatsLogger slog.LoggerInterface
```

Reply:
I changed it to `DogstatsdDebugLogger` so it would be easier to understand in the future